src/scenario/action_types/trade_price.c

```diff
@@ -0,0 +1,24 @@
+#include "trade_price.h"
+
+#include "city/message.h"
+#include "empire/trade_prices.h"
+
+int scenario_action_type_trade_price_adjust_execute(scenario_action_t *action)
+{
+    int32_t resource = action->parameter1;
+    int32_t adjustment = action->parameter2;
+    int32_t show_message = action->parameter3;
+
+    int successfully_changed = trade_price_change(resource, adjustment);
```

Comment:
You should also return here if `adjustment` is `0`.

Reply:
Done.